modules/st2-elastic-field/directive.js

```diff
@@ -0,0 +1,177 @@
+        if ($textarea.minRows) {
+          dimensions.minHeight = window.Math.max(dimensions.minHeight,
+                                                 $textarea.minRows * lineHeight + dimensions.outer.height);
+        }
+        
+        if ($textarea.maxRows) {
+          dimensions.maxHeight = window.Math.min(dimensions.maxHeight,
+                                                 $textarea.maxRows * lineHeight + dimensions.outer.height);
+        }
+        
+        return dimensions;
+      }
+      
+      function imitate(source, target) {
+        var computedStyle = window.getComputedStyle(source),
+            propertiesToCopy = [
+              'box-sizing',
+              'width',
+              'padding-top',
+              'padding-right',
+              'padding-bottom',
+              'padding-left',
+              'border-top-width',
+              'border-right-width',
+              'border-bottom-width',
+              'border-left-width',
+              'font-family',
+              'font-size',
+              'font-weight',
+              'font-style',
+              'letter-spacing',
+              'line-height',
+              'text-transform',
+              'word-spacing',
+              'text-indent'
+            ],
+            styleString = '';
+         
+        angular.forEach(propertiesToCopy, function(property) {
+          styleString += property + ':' + computedStyle.getPropertyValue(property) + ';';
+        });
+        
+        target.setAttribute('style', styleString);
+        target.value = source.value;
+      }
+      
+      function adjust($textarea, $clone, force) {
+        if ($textarea.isBeingAdjusted && !force) {
+          return;
+        }
+        
+        var dimensions = getDimensions($textarea),
+            overflow = 'hidden',
+            cloneHeight;
+        
+        $textarea.isBeingAdjusted = true;
+        imitate($textarea[0], $clone[0]);
+        cloneHeight = $clone[0].scrollHeight;
+        
+        if (cloneHeight > dimensions.maxHeight) {
+          cloneHeight = dimensions.maxHeight;
+          overflow = 'scroll';
+        } else if (cloneHeight < dimensions.minHeight) {
+          cloneHeight = dimensions.minHeight;
+        }
+        
+        $textarea[0].style.overflowY = overflow;
+
+        if (dimensions.height !== cloneHeight) {
+          $textarea[0].style.height = cloneHeight + 'px';
+        }
+          
+        $textarea.isBeingAdjusted = false;
+      }
+      
+      return {
+        restrict: 'C',
+        require: 'ngModel',
+        link: function (scope, $textarea, attrs, ngModel) {
+          if ($textarea[0].nodeName !== 'TEXTAREA' || !window.getComputedStyle || $textarea.data('elasticField')) {
+            return;
+          }
+          
+          
+          var $window = angular.element(window),
+              textarea = $textarea[0],
+              $clone = angular.element('<textarea class="st2-elastic-field-clone" tabindex="-1">'),
```

Comment:
We use scrollHeight to determine the height of the text in a textarea and then we set the height of the textarea itself equal to it. It would be a perfect property for the case if not one flaw: it is never less than textarea's height. That's why we use a hidden clone with zero height.
I don't know of any other approaches to the problem (except contenteditable, which is a whole another story). And I don't see how container's `overflow: hidden` can help here.